sklearn/utils/validation.py

```diff
@@ -103,6 +103,9 @@ def _is_arraylike(x):
 
 def _num_samples(x):
     """Return number of samples in array-like x."""
+    if hasattr(x, 'fit'):
+        # Don't get num_samples from an ensembles length!
+        raise TypeError('Expected dataset, but found estimator {!r}'.format(x))
```

Comment:
Python 2.6 doesn't like seem to understand this... Could you please use `{0!r}` instead?